pkg/csi/service/wcp/controller.go

```diff
@@ -975,9 +1161,19 @@ func (c *controller) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequ
 				return nil, csifault.CSIUnimplementedFault, logger.LogNewErrorCode(log, codes.Unimplemented,
 					"file volume feature is disabled on the cluster")
 			}
-
+			// Block file volume provisioning if FSS Workload_Domain_Isolation_Supported is enabled but
+			// 'fileVolumeActivated' field is set to false in vSphere config secret.
+			if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.WorkloadDomainIsolationSupported) &&
+				!c.manager.VcenterConfig.FileVolumeActivated {
+				return nil, csifault.CSIUnimplementedFault, logger.LogNewErrorCode(log, codes.Unimplemented,
+					"file volume service is disabled on cluster(s)")
```

Comment:
I guess file services are enabled on the supervisor level, is this error message appropriate? 
cc: @divyenpatel

Reply:
as per https://confluence.eng.vmware.com/pages/viewpage.action?spaceKey=CNAS&title=Continue+Support+RWM+Volumes+after+adding+new+workload+zone#ContinueSupportRWMVolumesafteraddingnewworkloadzone-UIChanges, file volume creation will be enabled on supervisor through the new toggle option i.e. fileVolumeActivated. 
If the message looks misleading, I will reword to "file volume creation is disabled on cluster(s)"

Reply:
I meant can we instead say `file services are disabled on supervisor cluster`. If we say cluster(s), they might search for that option on individual clusters.

Reply:
done, updated log message